File: day-1/day1.js
const fs = require("fs");
const arr = [];
let first;
let second;
let ans = 0;
let temp;
fs.readfile("day1.txt", "utf8", (err, data) => {
  if (err) {
    console.error("error reading file", err);
    return;
  }
  const lines = data.split("\n");
  arr.push(lines);

  for (let i = 0; i < arr.length; i++) {
    first = "";
    second = "";
    let str = arr[i];
    for (let j = 0; j < str.length; j++) {
      console.log(str.length);
      if (str[j] >= "0" && str[j] <= "9") {
        if (first == "") {
          first = str[j];
        } else {
          second = str[j];
        }
      }
    }
    if (second == "") {
      second = first;
    }
    temp = first + second;
    ans += parseint(temp);
  }
  console.log(ans);
});

// for (let i = 0; i < s.length; i++) {
//   if (s[i] >= "0" && s[i] <= "9") {
//     if (first == "") {
//       first = s[i];
//     } else {
//       second = s[i];
//     }
//   }
// }
// if (second == "") {
//   second = first;
// }
// temp = first + second;
// ans = parseInt(temp);

File: day-1/day1.cpp
#include <fstream>
#include <iostream>
#include <string>
#include <vector>

int main(int argc, char *argv[]) {
  // This is for accessing the data and storing into vectors.
  std::ifstream file("day1.txt");
  std::vector<std::string> lines;
  if (file.is_open()) {
    std::string line;
    while (std::getline(file, line)) {
      lines.push_back(line);
    }
    file.close();
  }
  // This is for accessing the stored data and calculating the value
  int ans = 0;
  for (int j = 0; j < lines.size(); j++) {
    std::string s = lines[j];
    std::string first = "";
    std::string second = "";
    int temp = 0;
    for (int i = 0; i < s.length(); i++) {
      if (isdigit(s[i])) {
        if (first == "") {
          first = s[i];
        } else {
          second = s[i];
        }
      }
    }
    if (second == "") {
      second = first;
    }
    temp = stoi(first + second);
    ans += temp;
  }
  std::cout << ans << std::endl;
  return 0;
}
